algebra/src/msm/variable_base.rs

```diff
@@ -205,32 +205,49 @@ impl VariableBaseMSM {
     where
         G::Projective: ProjectiveCurve<Affine = G>
     {
-        let scal_len = scalars.len();
+        let c = Self::get_optimal_window_size::<G>(scalars.len());
 
-        let c: usize = if scal_len < 32 {
+        Self::multi_scalar_mul_affine_c(bases, scalars, c)
+    }
+
+    fn get_optimal_window_size<G: AffineCurve>(scalars_len: usize) -> usize {
+        let c: usize = if scalars_len < 32 {
             3
         } else {
-            (2.0 / 3.0 * (f64::from(scalars.len() as u32)).log2() - 2.0).ceil() as usize
+            (2.0 / 3.0 * (f64::from(scalars_len as u32)).log2() - 2.0).ceil() as usize
         };
 
-        Self::multi_scalar_mul_affine_c(bases, scalars, c)
+        #[cfg(feature = "bn_382")]
+        if std::any::TypeId::of::<G>() == std::any::TypeId::of::<crate::curves::bn_382::G1Affine>()
+        || std::any::TypeId::of::<G>() == std::any::TypeId::of::<crate::curves::bn_382::G2Affine>()
+        || std::any::TypeId::of::<G>() == std::any::TypeId::of::<crate::curves::bn_382::g::Affine>()
+        {
+            return match scalars_len {
+                scalars_len if scalars_len <= 1 << 16 => c,
+                scalars_len if scalars_len <= 1 << 21 => 12,
+                scalars_len if scalars_len <= 1 << 23 => 16,
+                _ => c
```

Comment:
Do the window size result from our benches? If so, worth a comment.

Reply:
Done